sktime/transformers/series_as_features/signature_based/_checks.py

```diff
@@ -0,0 +1,63 @@
+"""
+_checks.py
+====================
+Contains a reusable decorator function to handle the sklearn signature checks.
+"""
+import functools
+import torch
+import pandas as pd
+from sktime.utils.validation.series_as_features import check_X, check_X_y
+from sktime.utils.data_container import nested_to_3d_numpy
+
+
+def handle_sktime_signatures(check_fitted=False):
```

Comment:
I see that the wrapper avoids some duplications, but it comes at the cost of readability I think, I'd prefer to have a few extra lines in the actual methods

Reply:
It works out being a lot of repeated code, my preference is to leave it in. I have added comments where the decorator is used to explain what is is doing though.